data/redirects.yml

```diff
@@ -85,6 +85,14 @@
   - /go/access-tokens/
 "/desktop/mac/apple-silicon/":
   - /go/apple-silicon/
+"/engine/api/#deprecated-api-versions":
+  - /engine/api/v1.23/
+  - /engine/api/v1.22/
+  - /engine/api/v1.21/
+  - /engine/api/v1.20/
+  - /engine/api/v1.19/
+  - /engine/api/v1.18/
```

Comment:
Hm... these don't work in previews, is that correct? Or is it because there's still a page available at this URL?

https://deploy-preview-19146--docsdocker.netlify.app/engine/api/v1.23/

Reply:
yeah preview deployments do not have cloudfront, which is what handles these pageless redirects. Hugo doesn't have a way to generate pages from data files, which we had a plugin for with Jekyll.

Reply:
we should change this to point to only the 1.24 markdown file instead of all md files

https://github.com/docker/docs/blob/main/hugo.yaml#L241-L243

Reply:
Isn't this redirecting `/engine/api/<old version>` to `/engine/api/#deprecated-api-versions` ? Or did I misunderstand your comment 😅

Reply:
No this is defining our imports from moby/moby and the target location to where we mount it in the virtual filesystem mount created by Hugo. So "*.md" files in docs/api are mounted to content/engine/api

Reply:
Is there anything I should change? (also feel free to push to my branch 😅)

Reply:
I tried removing those pages.. but we'll be getting broken links then.

```
Start building sites … 
hugo v0.121.2-6d5b44305eaa9d0a157946492a6f319da38de154+extended darwin/arm64 BuildDate=2024-01-05T12:21:15Z VendorInfo=brew

ERROR [en] REF_NOT_FOUND: Ref "v1.24.md" from page "engine/api/version-history.md": page not found
ERROR [en] REF_NOT_FOUND: Ref "v1.23.md" from page "engine/api/version-history.md": page not found
ERROR [en] REF_NOT_FOUND: Ref "v1.22.md" from page "engine/api/version-history.md": page not found
ERROR [en] REF_NOT_FOUND: Ref "v1.21.md" from page "engine/api/version-history.md": page not found
ERROR [en] REF_NOT_FOUND: Ref "v1.20.md" from page "engine/api/version-history.md": page not found
ERROR [en] REF_NOT_FOUND: Ref "v1.19.md" from page "engine/api/version-history.md": page not found
ERROR [en] REF_NOT_FOUND: Ref "v1.18.md" from page "engine/api/version-history.md": page not found
Built in 5417 ms
Error: error building site: logged 7 error(s)
```

Reply:
Ah, hm... yeah we need to take a look at that, and possibly remove them in upstream.

Do you know if redirects take priority over "page content" ? Because if that's the case, I guess people would just hit the redirect, and things should be fine.